nsxt/resource_nsxt_compute_manager.go

```diff
@@ -0,0 +1,531 @@
+		},
+		Schema: map[string]*schema.Schema{
+			"revision":     getRevisionSchema(),
+			"description":  getDescriptionSchema(),
+			"display_name": getDisplayNameSchema(),
+			"tag":          getTagsSchema(),
+			"access_level_for_oidc": {
+				Type:         schema.TypeString,
+				Description:  "Specifies access level to NSX from the compute manager",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(accessLevelForOidcValues, false),
+				Default:      model.ComputeManager_ACCESS_LEVEL_FOR_OIDC_FULL,
+			},
+			"create_service_account": {
+				Type:        schema.TypeBool,
+				Description: "Specifies whether service account is created or not on compute manager",
+				Optional:    true,
+				Default:     false,
+			},
+			"credential": {
+				Type:        schema.TypeList,
+				Description: "Login credentials for the compute manager",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"saml_login_credential": {
+							Type:        schema.TypeList,
+							Description: "A login credential specifying saml token",
+							Optional:    true,
+							MaxItems:    1,
+							ExactlyOneOf: []string{
+								"credential.0.saml_login_credential",
+								"credential.0.session_login_credential",
+								"credential.0.username_password_login_credential",
+								"credential.0.verifiable_asymmetric_login_credential",
+							},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"thumbprint": {
+										Type:        schema.TypeString,
+										Description: "Thumbprint of the server",
+										Optional:    true,
+									},
+									"token": {
+										Type:        schema.TypeString,
+										Description: "The saml token to login to server",
+										Optional:    true,
+										Sensitive:   true,
+									},
+								},
+							},
+						},
+						"session_login_credential": {
+							Type:        schema.TypeList,
+							Description: "A login credential specifying session_id",
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"session_id": {
+										Type:        schema.TypeString,
+										Description: "The session_id to login to server",
+										Optional:    true,
+										Sensitive:   true,
+									},
+									"thumbprint": {
+										Type:        schema.TypeString,
+										Description: "Thumbprint of the login server",
+										Optional:    true,
+									},
+								},
+							},
+						},
+						"username_password_login_credential": {
+							Type:        schema.TypeList,
+							Description: "A login credential specifying a username and password",
+							Optional:    true,
+							MaxItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"password": {
+										Type:        schema.TypeString,
+										Description: "The authentication password for login",
```

Comment:
I think `username` and `password` are required

Reply:
For some reason they aren't [listed](https://vdc-download.vmware.com/vmwb-repository/dcr-public/612caf56-5bb1-4a0b-8d2b-12ccd8981904/37be75ba-aa56-42f4-99c3-f6c989ddbe57/api_includes/types_UsernamePasswordLoginCredential.html) as required.
Yet the UI forces to specify these.

Reply:
Can we check this against NSX to see if empty creds are rejected? Same regarding other types of creds.

Reply:
I think that it's a waste of time - there's no way anyone could login without creds. I made these required already.